api/types/types.proto

```diff
@@ -3669,3 +3669,42 @@ message ConnectionDiagnosticTrace {
     // Error contains the low level error message in case of a failure.
     string Error = 5 [ (gogoproto.jsontag) = "error" ];
 }
+
+// AlertSeverity represents how problematic/urgent an alert is, and is used to assist
+// in sorting alerts for display.
+enum AlertSeverity {
+    LOW = 0;
+    MEDIUM = 5;
+    HIGH = 10;
+}
+
+// ClusterAlert is a cluster-level alert message.
+message ClusterAlert {
+    ResourceHeader Header = 1
+        [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "", (gogoproto.embed) = true ];
+    ClusterAlertSpec Spec = 2 [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "spec" ];
+}
+
+// ClusterAlertSpec is a cluster alert specification.
+message ClusterAlertSpec {
+    // Severity represents how problematic/urgent the alert is.
+    AlertSeverity Severity = 1 [ (gogoproto.jsontag) = "severity" ];
+    // Message is the user-facing message associated with the alert.
+    string Message = 2 [ (gogoproto.jsontag) = "message" ];
+    // Created is the time at which the alert was generated.
+    google.protobuf.Timestamp Created = 3 [
+        (gogoproto.jsontag) = "created,omitempty",
+        (gogoproto.stdtime) = true,
+        (gogoproto.nullable) = false
+    ];
+}
+
+// ClusterAlertQuery matches cluster alerts.
+message ClusterAlertQuery {
```

Comment:
What would you say renaming it to `GetClusterAlertsRequest` ?

Reply:
I'd say that I hate how unnecessarily long that name is, but its the right pattern to follow in this case lol.